src/utils/midi.ts

```diff
@@ -16,31 +15,37 @@ export interface MidiNote {
   volume: number
 }
 
-function loadInstrument(instrument: string): Promise<boolean> {
+function loadInstrument(instrument: string): void {
   const soundfontUrl = convertFileSrc('', 'midi')
 
-  return new Promise((resolve) => {
-    MIDI.loadPlugin({
-      instrument,
-      soundfontUrl,
-      targetFormat: 'mp3',
-      onerror: () => resolve(false),
-      onsuccess: () => {
-        loadedInstruments.add(instrument)
-
-        resolve(true)
-      },
-    })
-  })
+  loadedInstruments.set(
+    instrument,
+    new Promise((resolve) => {
+      MIDI.loadPlugin({
+        instrument,
+        soundfontUrl,
+        targetFormat: 'mp3',
+        onerror: () => {
+          loadedInstruments.delete(instrument)
+          resolve(false)
+        },
+        onsuccess: () => {
+          resolve(true)
+        },
+      })
+    }),
+  )
 }
 
 export async function playNote(note: MidiNote) {
   const wake = async () => await tauri.invoke('wake_sync')
 
-  if (!loadedInstruments.has(note.name)) {
-    if (!(await loadInstrument(note.name))) {
-      return await wake()
-    }
+  if (
+    !loadedInstruments.has(note.name) ||
+    !(await loadedInstruments.get(note.name))
```

Comment:
Are we using the value of undefined here? Still seems a bit strange.

Reply:
The promise stored in the map resolves to true or false